src/main/scala/nl.knaw.dans.easy.solr4files/ApplicationWiring.scala

```diff
@@ -58,36 +57,71 @@ class ApplicationWiring(configuration: Configuration)
       feedbackMessage <- updateFiles(bag, ddm, filesXML)
       _ <- commit()
     } yield feedbackMessage
+  }.recoverWith {
+    case t: SolrCommitException => Failure(t)
+    case t =>
+      commit()
+      Failure(t)
   }
 
-  def delete(bagId: String): Try[FeedBackMessage] = for {
-    _ <- deleteBag(bagId)
-    _ <- commit()
-  } yield s"Deleted file documents for bag $bagId"
+  def delete(bagId: String): Try[FeedBackMessage] = {
+    for {
+      _ <- deleteBag(bagId)
+      _ <- commit()
+    } yield s"Deleted file documents for bag $bagId"
+  }.recoverWith {
+    case t: SolrCommitException => Failure(t)
+    case t =>
+      commit()
```

Comment:
if this `commit()` returns a `Failure` as well, you'll loose it...

```scala
commit().flatMap(_ => Failure(t))
  .getOrRecover(t2 => new CompositeException(t2, t))
```

Reply:
tried variations but don't get it compiling.